### PR TITLE
add routes and form validation

### PR DESCRIPTION
**Pull request by Jachin-Manda**
- add a route for viewing properties
- add a link in the sidebar for managing and updating the user account information including uploading a profile picture
- add update and delete buttons for a property
- add form validation with flashing error messages if form input doesn't meet requirements
-

### CODE REVIEW

**Comment by Jachin-Manda** [comment]
Since Sqlite3 doesn't support storing ARRAY's on columns, I used a python dictionary. To save the list of image filenames, the filenames are added to a python dictionary object which is then converted to a JSON object and then saved to the database as a string. When querying the list of image filenames, the JSON object returned from the database is converted back to a python dictionary.

File: app/home/routes.py [hunk 1]
```diff
@@ -57,38 +77,126 @@ def get_segment(request):
...
+        return redirect(url_for('home_blueprint.view_property'))
 
-            file_path = os.path.join(current_app.root_path, 'base/static/property_images', clean_filename)
-            img_file.save(file_path)
+    elif request.method == 'GET':
+        # prefills the forms with the attributes to the property to be updated
+        form.prop_name.data = prop_to_update.name
+        form.prop_desc.data = prop_to_update.desc
+        form.prop_price.data = prop_to_update.price
+        form.prop_location.data = prop_to_update.location
```
Comment (sonlinux) [inline app/home/routes.py:188]: some pseudocode to consider for optimisation here, just for your thoughts 😛 
```
for data in form.data:
   form.item = property_to_update.data.item
   ```

**Review by sonlinux — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 2]
```diff
@@ -27,3 +27,8 @@ _templates
 # javascript
 package-lock.json
 .vscode/symbols.json
+
+# image files
+.jpeg
+.jpg
+.png
```
[the remark this answers is not in the document]
Comment (Jachin-Manda) [reply .gitignore:34]: Resolved

File: app/home/routes.py (hunk 1 above)
Comment (Jachin-Manda) [reply app/home/routes.py:188]: I am getting an error  'PropertyForm' object has no attribute 'items' in the browser

File: app/base/forms.py [hunk 3]
```diff
@@ -22,10 +24,43 @@ class CreateAccountForm(FlaskForm):
...
-    prop_name = StringField('Property Name', validators=[DataRequired()])
-    prop_desc = TextAreaField('Property Description', validators=[DataRequired(), Length(min=10, max=256)])
-    prop_photos = MultipleFileField('Upload photos of your property', validators=[FileAllowed(['jpg', 'png'])])
+    prop_name = StringField('Property Name', validators=[DataRequired(), Length(min=5, max=50)])
+    prop_desc = TextAreaField('Property Description', validators=[DataRequired(), Length(min=10, max=300)])
+    prop_photos = MultipleFileField('Upload photos of your property',
+                                    validators=[
+                                        DataRequired(),
+                                        FileAllowed(['jpg', 'png']),
+                                        Length(min=3, max=15, message="Upload 3 or more photos and not more that 15")])
```
[the remark this answers is not in the document]
Comment (Jachin-Manda) [reply app/base/forms.py:59]: Installed the tool and refactored code.

**Comment by sonlinux** [comment]
LGTM

**Review by sonlinux — APPROVED** [review]
LGTM

**Comment by Jachin-Manda** [comment]
Made proposed changes.

**Pull request merged**